openedx/core/djangoapps/programs/utils.py

```diff
@@ -127,6 +127,34 @@ def engaged_programs(self):
 
         return programs
 
+    def does_course_still_qualify_as_in_progress(self, now, enrolled_run_modes, course):
+        # Part 1: Check if any of the seats you are enrolled in qualify this course as in progress
+        enrolled_runs = [run for run in course['course_runs'] if run['key'] in self.course_run_ids]
+        # Check if you are enrolled in the required mode for the run
+        runs_with_required_mode = [
+            run for run in enrolled_runs
+            if run['type'] in enrolled_run_modes[run['key']]
```

Comment:
Don't use `in` here. It happens to work here, but what you really want is `==`; `enrolled_run_modes[run['key']]` is a string.